substra/sdk/graph.py

```diff
@@ -82,15 +82,15 @@ def compute_ranks(
 
     edges = set()
 
-    while len(visited) != n_nodes:
+    while len(visited) != len(node_graph):
         current_node = _get_current_node(visited, ranks)
         visited.add(current_node)
         for child in inverted_node_graph.get(current_node, list()):
             ranks[child] = max(ranks[current_node] + 1, ranks.get(child, -1))
 
             # Cycle detection
             edge = (current_node, child)
-            if (edge[1], edge[0]) in edges:  # if edge in edges or
+            if (edge[1], edge[0]) in edges:
                 raise exceptions.InvalidRequest(
                     "missing dependency among inModels IDs", 400
```

Comment:
We could use a very specific message here, maybe `Circular dependency among tuples between {edge[0]} and edge[1]`

Reply:
My problem is that in substra-tests, this test: `test_compute_plan_circular_dependency_failure` asserts that the error message is the same as the one returned by the backend

```python
with pytest.raises(substra.exceptions.InvalidRequest) as e:
        client.add_compute_plan(cp_spec)

assert 'missing dependency among inModels IDs' in str(e.value)
```

so I can add the edges at fault to the message but I need to keep the rest of the exception as is or change the test. I would like to keep something close to what the backend returns too (maybe we should return a clearer message from the backend, this one seems to refer to a different problem).

Reply:
Agreed, 👍